dips/dip-158.md

```diff
@@ -0,0 +1,763 @@
+
+```
+"receiver": {
+  "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+  "business_data": {
+    BusinessDataObject()
+  }        
+}
+```
+## PaymentSenderObject
+A PaymentSenderObject represents the person in the payment. In P2M scenarios it is the sender of funds.
+
+|Field|Type|Required?|Description|
+|-|-|-|-|
+| address | str | Y | Address of the receiver account. The addresses are encoded using bech32| 
+|payer_data|[PayerDataObject](#payerdataobject)|Y|The details of the payer|
+
+```
+"sender": {
+  "address": "dm1pg9q5zs2pg9q5zs2pg9q5zs2pg9skzctpv9skzcg9kmwta",
+  "payer_data": {
+    PayerDataObject() 
+  }          
+}
+```
+
+## PayerDataObject
+The details of the payer. payer_data is the minimal set of payer information required by the merchant to perform risk checks and apply AML policies.  
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|given_name|str|Y|Legal given name of the payer|
+|surname|str|Y|Legal surname of the payer|
+|address|AddressObject|Y|Merchant's physical address information|
+|national_id|[NationalIdObject](https://github.com/diem/dip/blob/master/dips/dip-1.mdx#nationalidobject)|N|National ID information for payer. e.g. SSN|
+
+```
+"payer_data": 
+{
+  "name": "Alice",
+  "surname": "Cooper",
+  "address": {
+    AddressObject()
+  },
+  "national_id": {
+    NationalIdObject()
+  }        
+}
+``` 
+## BusinessDataObject
+The details of the merchant.
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|name|str|Y|Merchant's display name. Should be recognizable by the payer|
+|legal_name|str|Y|The legal entity name|
+|address|[AddressObject](https://github.com/diem/dip/blob/master/dips/dip-1.mdx#addressobject)|Y|Merchant's physical address information|
+```
+"business_data": 
+{
+  "name": "Acme",
+  "legal_name": "Acme Technologies Inc.",
+  "address": {
+    AddressObject()
+  }        
+}
+``` 
+
+## PaymentActionObject
+Information about the payment type (i.e. action), amount and currency.
+
+|Field|Type|Required?|Description|
+|-----|----|---------|-----------|
+| amount | uint | Y | Amount of the transfer.  Base units are the same as for on-chain transactions for this currency.  For example, if DiemUSD is represented on-chain where "1" equals 1e-6 dollars, then "1" equals the same amount here.  For any currency, the on-chain mapping must be used for amounts. |
+| currency | enum | Y | One of the supported on-chain currency types, e.g. XUS|
+| action | [Action enum](#action-enum) | Y | Populated in the request.  This value indicates the requested action to perform. For immediate capture and P2P transfer, "charge" is used.  For auth and capture, "auth" and "capture" should be used. "capture" can only be performed after a valid "auth" |
+| valid_until | uint | N | Unix timestamp indicating until when the payment can be authorized. Once this time has been reached, the payment cannot be authorized and funds cannot be locked. |
```

Comment:
These are 2 separate issues: 

Re 1 (for how long funds are locked?):
Authorization is usually valid for 30 days. However, it can be less or more (usually no longer than 90 days).
In standard industry, it is up to the payment method or issuer to determine the period. In Diem's case, it should be the sending wallet that should determine it. I don't think we should force anything in the protocol.
If a capture request is sent after the authorization period elapsed, the capture will fail as the funds are no longer locked.

Re 2 (valid_until):
valid_until is used for determining when is the latest time the funds lock can **start**. This will usually be a much shorter period, e.g. 30 minutes in which the user must approve the payment based on the provided details. It is up to the merchant to decide how long it is willing to wait for the user approval.
At First, we allow the merchant to specify the aforementioned period as part of the REST API we provide the PSP. If no value is provided when the payment is created, we set 30 minutes as default that proliferates to the valid_until element.

Reply:
Can we give guidelines for other VASPs to have sane defaults? A lot of the values mentioned above (auth valid for 30 days, ability to auth valid_until 30 minutes) aren't discoverable at all and could leave an implementing VASP confused.

Reply:
Sure. I will add that the recommended default for valid_until is 30 minutes (assuming a value was not provided by the merchant).

Regarding the "recommended" 30 days in which the authorization can be captured, I prefer not to add anything at this stage as this is part of the VASP business logic and not part of the protocol (and not mentioned in the DIP at all).